tracer/src/Datadog.Tracer.Native/CMakeLists.txt

```diff
@@ -11,6 +11,31 @@ project("Datadog.Tracer.Native" VERSION 2.20.0)
 # Environment detection
 # ******************************************************
 
+SET(OSX_ARCH ${CMAKE_OSX_ARCHITECTURES})
+
+# Detect architecture
+if (OSX_ARCH STREQUAL x86_64)
+    message(STATUS "Architecture is x64/AMD64 configured by CMAKE_OSX_ARCHITECTURES")
+    SET(ISAMD64 true)
+elseif (OSX_ARCH STREQUAL arm64)
+    message(STATUS "Architecture is ARM64 configured by CMAKE_OSX_ARCHITECTURES")
+    SET(ISARM64 true)
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
+    message(STATUS "Architecture is x64/AMD64")
+    SET(ISAMD64 true)
+    SET(OSX_ARCH "x86_64")
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL x86 OR CMAKE_SYSTEM_PROCESSOR STREQUAL i686)
+    message(STATUS "Architecture is x86")
+    SET(ISX86 true)
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL arm64)
+    message(STATUS "Architecture is ARM64")
+    SET(ISARM64 true)
+    SET(OSX_ARCH "arm64")
+elseif (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l OR CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
+    message(STATUS "Architecture is ARM")
+    SET(ISARM true)
+endif()
+
```

Comment:
if it were added in the top level one, we wouldn't have to duplicate it here I think